api.planx.uk/modules/auth/service.ts

```diff
@@ -59,3 +60,16 @@ const getAllowedRolesForUser = (user: User): Role[] => {
 const getDefaultRoleForUser = (user: User): Role => {
   return user.isPlatformAdmin ? "platformAdmin" : "teamEditor";
 };
+
+/**
+ * A staging-only user cannot access production, but can access all other envs
+ */
+export const checkUserCanAccessEnv = async (
+  email: string,
+  env?: string,
+): Promise<boolean> => {
+  const isStagingOnlyUser = await $api.user.isStagingOnly(email);
```

Comment:
**Why is this it it's own function and not a field added to `user.getByEmail()`**

I appreciate that this does mean an additional query but the `user.isStagingOnly` property is only ever needed here once on login.

If we added this to the more general query, we'd be polluting our `User` type with a meaningless property that has no value outside of this context.

This is one of those cases where the full database model will not match the model we want in-application.

Reply:
Makes sesne to me, thanks for explanation!